src/main/java/com/aws/greengrass/config/ConfigurationReader.java

```diff
@@ -97,9 +115,45 @@ public static void mergeTLogInto(Configuration config, Path tlogPath, boolean fo
         }
     }
 
-    private static void mergeTLogInto(Configuration c, Path p) throws IOException {
+    private static void mergeTLogInto(Configuration c, Path p, ConfigurationMode configurationMode) throws IOException {
         try (BufferedReader bufferedReader = Files.newBufferedReader(p)) {
-            mergeTLogInto(c, bufferedReader, false, null);
+            mergeTLogInto(c, bufferedReader, false, null, configurationMode);
+        }
+    }
+
+    /**
+     * In place update for the given config object from the given transaction log, adhering to the given update behavior
+     * tree and without losing listeners.
+     *
+     * @param config         configuration to merge into
+     * @param tlogPath       path of the tlog file to read to-be-replaced config from
+     * @param forceTimestamp should ignore if the proposed timestamp is older than current
+     * @param mergeCondition Predicate that returns true if the provided Topic should be merged and false if not
+     * @param tree           Merge behavior hierarchy for the update
+     * @throws IOException if update fails
+     */
+    public static void updateFromTLog(Configuration config, Path tlogPath, boolean forceTimestamp,
+                                      Predicate<Node> mergeCondition, UpdateBehaviorTree tree) throws IOException {
+
+        // Merge tlog into configuration so that nodes to retain get replaced
+        ConfigurationReader.mergeTLogInto(config, tlogPath, forceTimestamp, mergeCondition);
+
+        // Remove extraneous nodes from configuration
```

Comment:
can you add context here about what type of nodes could remain in configuration? This comment doesn't tell me anything more than what the "discardNodesNotInTlog" does